src/extensions/custom-view/bootstrap-table-custom-view.js

```diff
@@ -0,0 +1,113 @@
+ * @github: https://github.com/UtechtDustin
+ */
+
+const Utils = $.fn.bootstrapTable.utils
+
+$.extend($.fn.bootstrapTable.defaults, {
+  customView: false,
+  showCustomView: false,
+  showCustomViewButton: false
+})
+
+$.extend($.fn.bootstrapTable.defaults.icons, {
+  customView: {
+    bootstrap3: 'glyphicon glyphicon-eye-open',
+    bootstrap4: 'fa fa-eye',
+    semantic: 'fa fa-eye',
+    foundation: 'fa fa-eye',
+    bulma: 'fa fa-eye',
+    materialize: 'remove_red_eye'
+  }[$.fn.bootstrapTable.theme] || 'fa-eye'
+})
+
+$.extend($.fn.bootstrapTable.defaults, {
+  onCustomViewPostBody () {
+    return false
+  },
+  onCustomViewPreBody () {
+    return false
+  }
+})
+
+$.extend($.fn.bootstrapTable.locales, {
+  formatToggleCustomView () {
+    return 'Toggle custom view'
+  }
+})
+$.extend($.fn.bootstrapTable.defaults, $.fn.bootstrapTable.locales)
+
+$.fn.bootstrapTable.methods.push('toggleCustomView')
+
+$.extend($.fn.bootstrapTable.Constructor.EVENTS, {
+  'custom-view-post-body.bs.table': 'onCustomViewPostBody',
+  'custom-view-pre-body.bs.table': 'onCustomViewPreBody'
+})
+
+$.BootstrapTable = class extends $.BootstrapTable {
+
+  init () {
+    if (this.options.showCustomView) {
+      this.showCustomView = this.options.showCustomView
+    }
+
+    super.init()
+  }
+
+  initToolbar (...args) {
+    super.initToolbar(...args)
+
+    if (this.options.customView && this.options.showCustomViewButton) {
+      const $btnGroup = this.$toolbar.find('>.' + this.constants.classes.buttonsGroup.split(' ').join('.')).first()
+      let $btnToggleCustomView = $btnGroup.find('.toggle-custom-view')
+
+      if (!$btnToggleCustomView.length) {
+        $btnToggleCustomView = $(`
+          <button class="toggle-custom-view ${this.constants.buttonsClass}"
+          type="button" title="${this.options.formatToggleCustomView()}">
+          ${this.options.showButtonIcons ? Utils.sprintf(this.constants.html.icon, this.options.iconsPrefix, this.options.icons.customView) : ''}
+          ${this.options.showButtonText ? this.options.formatToggleCustomView() : ''}
+          </button>
+        `).appendTo($btnGroup)
+
+        $btnToggleCustomView.on('click', $.proxy(this.toggleCustomView, this))
+      }
+    }
+  }
+
+  initBody () {
+    super.initBody()
+    const $table = this.$el
+    const $customViewContainer = this.$container.find('.fixed-table-custom-view')
+
+    $table.hide()
+    $customViewContainer.hide()
+    if (!this.options.customView || !this.showCustomView) {
+      $table.show()
+      return
+    }
+
+    const _data = this.getData()
+    const data = []
+    for (let i = this.pageFrom - 1; i < this.pageTo; i++) {
+      data.push(_data[i])
+    }
+
+    const value = Utils.calculateObjectValue(this, this.options.customView, [data], '')
+    this.trigger('custom-view-pre-body', data, value)
+    if ($customViewContainer.length === 1) {
+      $customViewContainer.show()
+      $customViewContainer.html(value)
```

Comment:
```js
if ($customViewContainer.length) {
  $customViewContainer.show().html(value)
}
```
```

Reply:
I dont like the method call behind method call syntax, but i changed it :-)